---
bool success
string message
WifiMessage[] observations
